easybuild/easyconfigs/l/lagrangian-filtering/lagrangian-filtering-0.8.3-foss-2022a.eb

```diff
@@ -0,0 +1,78 @@
+# This installation unfortunately requires the GitHub dance, as the tarball
+# is simply not installing. For that reason there is no checksum! 
+# Author: J. Sassmannshausen (Imperial College London/UK)
+
+easyblock = 'PythonBundle'
+
+name = 'lagrangian-filtering'
+version = '0.8.3'
+
+homepage = 'https://github.com/angus-g/lagrangian-filtering'
+description = """Temporal filtering of data in a Lagrangian frame of reference.""" 
+
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+dependencies = [
+    ('Python', '3.10.4'),
+    ('dask', '2022.10.0'),
+    ('h5py', '3.7.0'),
+    ('SciPy-bundle', '2022.05'),
+    ('netcdf4-python', '1.6.1'),
+    ('scikit-learn', '1.1.2'),
+    ('tqdm', '4.64.0'),
+    ('zarr', '2.13.3'),
```

Comment:
I cannot find any requirement of `lagrangian-filtering` or its extensions on these dependencies. Can you tell me where this come from, or are added for convenience?

Reply:
@lexming https://github.com/angus-g/lagrangian-filtering/blob/master/setup.py#L11

Reply:
Hmm, when I made that comment GitHub was showing me more than just those 3 lines as highlighted. Disregard

Reply:
huh, `parcels` looks like one of the few packages that actually uses `mpi4py` as well

Reply:
`parcels` conda dependency list: https://github.com/OceanParcels/parcels/blob/master/environment_py3_linux.yml

Reply:
I see, that explains why I did not find these dependencies. Parcels has many dependencies but it does not define any `install_requires`, so `pip` is happy with anything and only `import` fails. The issue is that the base import does not show all possible failures, as it has even more deps which are only listed in that file with the conda environment.

Reply:
The more I look into this, the more I think that it would be better to put Parcels in its own easyconfig. What do you think?

Reply:
Good question. My gut-feeling is that having `parcel` as a separate package might be easier to maintain. From what I could see, only `gdc-client` is using it and that is only up to `foss-2018b`. That said, the `gdc-client-1.6.0-GCCcore-10.2.0.eb` version does not have that any more. Also, the latest release is from 2013 (sic!) and the homepage is gone. So frankly: does a separate EasyConfig file makes sense here from the prospective it is old software? In some respect it would be better to raise an issue with the developers of `lagrangian-filtering` and ask them to move on.
I am open for discussion here. :-)

Reply:
ERRATA:
The homepage on Pypi for `parcle` seems to be outdated! 
I just found the [new](https://github.com/angus-g/parcels/tags) one. 
In that case, I would suggest to go for a separate EasyConfig file. I think I relied too much on Pypi being the source of all Python packages. :-)

Reply:
@sassy-crick I made the easyconfig for `Parcels`, see https://github.com/easybuilders/easybuild-easyconfigs/pull/16838 . Please update this PR for `lagrangian-filtering` to depend on parcels instead of bundling it.

Reply:
Ok, now we got another 'issue'. We already had a merged `xarray` (PR #16685) which is version 2022.9.0 but your PR #16838 is using version 2022.6.0. What is the best way forward here?
I am happy to downgrade to your version, I just want to avoid more problems further along the line. :-)

Reply:
Even though we merged version 2022.9.0, we have to stick to version 2022.6.0 because there are already other easyconfigs in this toolchain using that lower version.

Reply:
No problem. I have used the `Parcel` version from the PR above, which means I could remove some Python extensions as well as they were already in that PR. 
I hope it is all ok now. :-)